geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/metadata/TableBasedMetadata.scala

```diff
@@ -231,6 +231,11 @@ trait TableBasedMetadata[T] extends GeoMesaMetadata[T] with LazyLogging {
   // checks that the table is already created, and creates it if not
   def ensureTableExists(): Unit = tableExists.set(true, false, createTable())
 
+
+  def resetCache():Unit={
+    tableExists = if (checkIfTableExists) { new NotSynchronized[Boolean](true)} else {new IsSynchronized[Boolean](false)}
```

Comment:
per my above comment, we could move this method out to `GeoMesaMetadata` with a no-op default implementation, and then override it here.

this piece is potentially problematic though... making tableExists into a var makes it not thread safe anymore. we should probably just always synchronize the value. the `MaybeSynchronized` code here was an attempt to avoid synchronization calls in the common case of the table already existing, assuming that it wouldn't be deleted ever, but that doesn't hold when `delete` is called on the data store.